### DISCUSSION TITLE
Set a custom header to all OUTGOING REQUESTS

### DISCUSSION THREAD

**Discussion by hosseinsarshar** [vercel/next.js#44253] [Ideas]
Hi,

Next JS config file allows you to set custom headers to the `response` of the incoming requests. However, I couldn't find any example on how to add custom headers to all `outgoing requests`. Let's say you need to append a custom header to retrieve images from a third-party provider, to the best of my knowledge you can't set headers to the Image component, therefore, you need another way to append a customer header to all outgoing requests so that the get request initiated from the Image component can pass the customer header requirement from the provider.

Help would be kindly appreciated

**Comment by ciruz** [comment]
I'm pretty sure you need to use ``fetch`` to request your image, where you can add all the custom headers you want and then load the result into an image (`Response.blob()` and  `URL.createObjectURL` => https://developer.mozilla.org/en-US/docs/Web/API/Response/blob#examples)

**Reply by hosseinsarshar** [reply]
I'm concerned about the performance of this method, the client needs download all the images in advance and you won't be able to leverage the turnkey performance gains of the Image component. You need to manually handle the lazy load and rebuild the Image component. Ideally, there needs to be a middleware that you'd be able to interject all/targeted outgoing requests and add a header tag to them, similar to how nextjs handles this for all client responses.

**Comment by GabenGar** [comment]
The client code in browser (and therefore `<Image>` component) can only set a limited subset of headers. You'd have to create a proxy "endpoint" using [rewrites](https://nextjs.org/docs/api-reference/next.config.js/rewrites), although I am not sure how can you pass headers this way.

**Comment by jarvislin94** [comment]
I think the only way to achieve this is using the `Middle`.
https://nextjs.org/docs/pages/building-your-application/routing/middleware
